stdlib/src/main/scala/chisel3/std/tilelink/Data.scala

```diff
@@ -0,0 +1,101 @@
+package chisel3.std.tilelink
+
+import chisel3._
+import chisel3.util.DecoupledIO
+
+import scala.collection.immutable.SeqMap
+
+private class NoTLCException(
+  channel:           String,
+  tlBundleParameter: TLBundleParameter)
+    extends ChiselException(
+      s"call $channel in TLBundle is not present in a TL-UL or TL-UH bus:\n $tlBundleParameter"
+    )
+
+class TLBundle(val parameter: TLBundleParameter) extends Record with experimental.AutoCloneType {
```

Comment:
I think `Record` is OK, since it's public API, there is no reason to forbid user or library designer using them...
I tried to implement a Scala type check version of tilelink library, including use `ChiselEnum`, however it turns to be over-design...